### DISCUSSION TITLE
Why the choice of returning a string when api call fails?

### DISCUSSION THREAD

**Discussion by fjrevoredo** [dynomake/libretranslate-java#7] [Q&A] [answered]
Hi there! the SDK is pretty nice so far. 
But I've noticed some weird behavior of the SDK that I cannot understand. Why when the translation fails instead of throwing an exception, the translate method returns a translation with "Falled translate!"??
There is no way of telling what went wrong(Connection fail? Libre translate instance issue? etc) and it's hard to even notice that something went wrong. 

My libretranslate instance was down and I never noticed something wrong until every translation got that message.

Thanks!

**Comment by dynomake** [comment]
Hi! Thanks for the feedback, I didn't even think about it. I'll make exceptions soon.

**Comment by dynomake** [comment, marked as answer]
I did it. Now an exception will be thrown if there are problems.

_Thank you for your ask._

**Reply by fjrevoredo** [reply]
Perfect! Thank you very much
